exchange/orderbooks_test.go

```diff
@@ -0,0 +1,108 @@
+package exchange
+
+import (
+	"testing"
+	"time"
+
+	"github.com/shopspring/decimal"
+)
+
+func testMarketRecord() MarketRecord {
+	return MarketRecord{
+		Timestamp: time.Now(),
+		Symbol:    "SKY/BTC",
+		Asks: []MarketOrder{
+			// this person is trying to sell 9 skycoins at a price of 5
+			// bitcoins per skycoin
+			// if their order is fulfilled, they lose 9 SKY and gain 45 BTC
+			{
+				Price:  decimal.NewFromFloat(5.0),
+				Volume: decimal.NewFromFloat(9.0),
+			},
+			// this person is trying to sell 8 skycoins at a price of 4
+			// bitcoins per skycoin
+			// if their order is fulfilled, they lose 8 SKY and gain 32 BTC
+			{
+				Price:  decimal.NewFromFloat(4.0),
+				Volume: decimal.NewFromFloat(8.0),
+			},
+		},
+		Bids: []MarketOrder{
+			// this person is trying to buy 7 skycoins at a price of 3
+			// bitcoins per skycoin
+			// if their order is fulfilled, they gain 7 SKY and lose 21 BTC
+			{
+				Price:  decimal.NewFromFloat(3.0),
+				Volume: decimal.NewFromFloat(7.0),
+			},
+			// this person is trying to buy 6 skycoins at a price of 2
+			// bitcoins per skycoin
+			// if their order is fulfilled, they gain 6 SKY and lose 12 BTC
+			{
+				Price:  decimal.NewFromFloat(2.0),
+				Volume: decimal.NewFromFloat(6.0),
+			},
+		},
+	}
+}
+
+func TestSpendItAll_Success(t *testing.T) {
```

Comment:
Add `testify/require` as a dependency and use it (see tests in skycoin/skycoin, skycoin/teller etc for usage)